### ISSUE TITLE
Create test file in non-existing directories

### ISSUE THREAD

**Issue by EmmaTye** [seancorfield/readme#4]
## Problem

Currently, generated filenames have to include existing directories. So if the directory `foo/bar` didn't exist, and I wanted to specify my test file as `foo/bar/baz.clj`, it would give the error
```
java.io.FileNotFoundException: foo/bar/baz.clj (No such file or directory)
```

## Fix

In [readme.clj](https://github.com/seancorfield/readme/blob/develop/src/seancorfield/readme.clj#L88), it `spit`s the generated file into `readme-test`. Before `spit`ting, run `makeparents` on the filename to add any non-existing directories
```clojure
(makeparents readme-test)
```

**Comment by seancorfield** [comment]
Thank you! Note that any such directories created will not be deleted, even though the generated test file itself will be deleted at the end of the test run.

**Comment by seancorfield** [comment]
This is fixed in 1.0.15. Then I released 1.0.16 with an additional bug I just noticed.
